workspace_server/src/main/java/gov/nasa/jpl/aerie/workspace/server/WorkspaceBindings.java

```diff
@@ -213,25 +215,27 @@ private void put(Context context) throws NoSuchWorkspaceException, IOException {
       type = context.queryParamAsClass("type", String.class)
                     .allowNullable()
                     .check(Objects::nonNull, "'type' must be provided.")
-                    .check(ts -> ts.equalsIgnoreCase("file") || ts.equalsIgnoreCase("directory"),
+                    .check(ts -> ts == null || ts.equalsIgnoreCase("file") || ts.equalsIgnoreCase("directory"),
```

Comment:
`ts` physically cannot be null, see L217 `.check(Objects::nonNull, "'type' must be provided.")`. Please remove this change or explain why it is necessary.

Reply:
I included this because this was throwing a failure when performing the `equalsIgnoreCase` check. I'm confused by this as presumably the above `check()` would short-circuit, but I guess the behavior of this is a bit more nuanced than I thought.

Reply:
Well, either way, `null` is **not** a permitted value. If an exception is occurring when `ts` is `null`, why not just reverse the check instead?

```suggestion
                    .check(ts -> "file".equalsIgnoreCase(ts) || "directory".equalsIgnoreCase(ts),
```